_posts/2018-02-15-totw-88.md

```diff
@@ -0,0 +1,152 @@
+
+* **Use assignment syntax when initializing directly with the intended literal
+  value (for example: ints, floats, strings), with smart pointers such as
+  `std::shared_ptr`, `std::unique_ptr`, with containers (`std::vector`,
+  `std::map`, etc), when performing struct initialization, or doing copy
+  construction.**
+
+  ```cpp
+  int x = 2;
+  string foo = "Hello World";
+  std::vector<int> v = {1, 2, 3};
+  std::unique_ptr<Matrix> matrix = NewMatrix(rows, cols);
+  MyStruct x = {true, 5.0};
+  MyProto copied_proto = original_proto;
+  ```
+
+  instead of:
+
+  ```cpp
+  int x{2};
+  string foo{"Hello World"};
+  std::vector<int> v{1, 2, 3};
+  std::unique_ptr<Matrix> matrix{NewMatrix(rows, cols)};
+  MyStruct x{true, 5.0};
+  MyProto copied_proto{original_proto};
+  ```
+   
+* **Use the traditional constructor syntax (with parentheses) when the
+  initialization is performing some active logic, rather than simply composing
+  values together.**
+
+  ```cpp
+  Frobber frobber(size, &bazzer_to_duplicate);
+  EventManager event_manager(options);
+  std::vector<double> fifty_pies(50, 3.14);
+  ```
+
+  vs.
+
+  ```cpp
+  Frobber frobber{size, &bazzer_to_duplicate};
+  ```
+  
+* **Use {} initialization without the = only if the above options don’t
+  compile:**
+
+  ```cpp
+  class Foo {
+   public:
+    Foo(int a, int b, int c) : array_{a, b, c} {}
+
+   private:
+    int array_[5];
+    // Requires {}s because the constructor is marked explicit
+    // and the type is non-copyable.
+    EventManager em{EventManager::Options()};
+  };
+  ```
+
+* **Never mix {}s and auto.**<br/>
+  For example, don’t do this:
+
+  ```cpp
+  // Bad code
+  auto x{1};
+  auto y = {2}; // This is a std::initializer_list<int>!
+  ```
+
+  (For the language lawyers: prefer copy-initialization over
+  direct-initialization when available, and use parentheses over curly braces
+  when resorting to direct-initialization.)
+
+Perhaps the best overall description of the issue is Herb Sutter’s
+[GotW post](http://herbsutter.com/2013/05/09/gotw-1-solution/). Although he
+shows examples that include direct initialization of `int` with braces, his
+final advice is roughly compatible with what we present here with one caveat:
+where Herb says "where you prefer to see only the = sign", we unambiguously
+prefer to see exactly that.
+
+## Conclusion
+
+The tradeoffs for uniform initialization syntax are not generally worth it: our
+compilers already warn against the Most Vexing Parse (you can use brace
+initialization or add parens to resolve the issue), and the safety from
+narrowing conversions isn’t worth the readability hit for brace-initialization
+(we’ll need a different solution for narrowing conversions, eventually). The
+Style Arbiters don’t think this issue is critical enough to make a formal rule
+on, especially because there are cases (notably in generic code) where brace
+initialization may be justified. But as general guidance, we don’t believe the
```

Comment:
Maybe drop the last sentence?

Reply:
Done in https://github.com/abseil/abseil.github.io/pull/145